common/djangoapps/track/backends/oli.py

```diff
@@ -78,10 +82,18 @@ def send(self, event):
             LOG.info('event_data attribute missing from event for OLI service')
             return None
 
+        problem_text = None
+        # Look where it should be for a capa prob.
         problem_id = event_data.get('problem_id')
+        if problem_id:
+            problem_text = self.get_problem_text(problem_id, course_id)
+
         if not problem_id:
-            LOG.info('problem_id attribute missing from event for OLI service')
-            return None
+            # Look where it should be for an xblock.
+            problem_id = context.get('module').get('usage_key')
```

Comment:
So it looks like if we need to grab the `problem_id` from here,
we don't refetch the `problem_text`; should we?